testing-unit-common/src/main/java/com/bumble/appyx/testing/unit/common/util/InteropSimpleBuilderStub.kt

```diff
@@ -0,0 +1,31 @@
+package com.bumble.appyx.testing.unit.common.util
+
+import androidx.lifecycle.Lifecycle
+import com.bumble.appyx.core.builder.SimpleBuilder
+import com.bumble.appyx.core.modality.BuildContext
+import com.bumble.appyx.core.node.Node
+import kotlin.test.assertEquals
+import kotlin.test.assertTrue
+
+class InteropSimpleBuilderStub(
+    val delegate: (BuildContext) -> Node = { _ -> NodeStub() },
+) : SimpleBuilder() {
+
+    private val _createdNodes = ArrayList<Node>()
+
+    val last: Node?
+        get() = _createdNodes.lastOrNull()
+
+    override fun build(buildContext: BuildContext): Node =
+        delegate(buildContext).also {
+            _createdNodes += it
+        }
+
+    fun assertCreatedNode() {
+        assertTrue(last != null, "Has not created any node")
+    }
+
+    fun assertLastNodeState(state: Lifecycle.State) {
+        assertEquals(state, last?.lifecycle?.currentState)
```

Comment:
for now if `last?.lifecycle?.currentState` returns null, you know that the last node was null. however potentially the API could change in the future which may make this test have a false positive/negative (i.e. how lifecycle is nullable, but the last node was not null, making it confusing to tell why the test failed, or worse it passes).

I proposed the following (because we don't seem to have soft asserts):
```
assertCreatedNode()
assertEquals(state, last!!.lifecycle.currentState)
```

This way if the API of lifecycle changes to be nullable, the code would not compile.

Reply:
I like it, thanks!
Changed in 592c9d9268cfd05127ea0c4739787a7430fd4fcc and 8cad9d2c97c70f01fe62e2a4b1ed603319dcd022